app/src/main/java/org/connectbot/TerminalView.java

```diff
@@ -150,6 +170,282 @@ public TerminalView(Context context, TerminalBridge bridge) {
+			return true;
+		} else if (event.getAction() == MotionEvent.ACTION_MOVE) {
+			int buttonState = event.getButtonState();
+			int button = (buttonState & MotionEvent.BUTTON_PRIMARY) != 0 ? 0 :
+				(buttonState & MotionEvent.BUTTON_SECONDARY) != 0 ? 1 :
+				(buttonState & MotionEvent.BUTTON_TERTIARY) != 0 ? 2 : 3;
+			((vt320) bridge.buffer).mouseMoved(
+				button,
+				col,
+				row,
+				(meta & KeyEvent.META_CTRL_ON) != 0,
+				(meta & KeyEvent.META_SHIFT_ON) != 0,
+				(meta & KeyEvent.META_META_ON) != 0);
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Takes an android mouse event and produces a Java InputEvent modifiers int which can be
+	 * passed to vt320.
+	 * @param mouseEvent The {@link MotionEvent} which should be a mouse click or release.
+	 * @return A Java InputEvent modifier int. See
+	 * http://docs.oracle.com/javase/7/docs/api/java/awt/event/InputEvent.html
+	 */
+	@TargetApi(14)
+	private static int mouseEventToJavaModifiers(MotionEvent mouseEvent) {
+		if (MotionEventCompat.getSource(mouseEvent) != InputDevice.SOURCE_MOUSE) return 0;
+
+		int mods = 0;
+
+		// See http://docs.oracle.com/javase/7/docs/api/constant-values.html
+		int buttonState = mouseEvent.getButtonState();
+		if ((buttonState & MotionEvent.BUTTON_PRIMARY) != 0)
+			mods |= 16;
+		if ((buttonState & MotionEvent.BUTTON_SECONDARY) != 0)
+			mods |= 8;
+		if ((buttonState & MotionEvent.BUTTON_TERTIARY) != 0)
+			mods |= 4;
+
+		// Note: Meta and Ctrl are intentionally swapped here to keep logic in vt320 simple.
+		int meta = mouseEvent.getMetaState();
+		if ((meta & KeyEvent.META_META_ON) != 0)
+			mods |= 2;
+		if ((meta & KeyEvent.META_SHIFT_ON) != 0)
+			mods |= 1;
+		if ((meta & KeyEvent.META_CTRL_ON) != 0)
+			mods |= 4;
+
+		return mods;
+	}
+
+	@Override
+	@TargetApi(12)
+	public boolean onGenericMotionEvent(MotionEvent event) {
+		if ((MotionEventCompat.getSource(event) & InputDevice.SOURCE_CLASS_POINTER) != 0) {
+			switch (event.getAction()) {
+			case MotionEvent.ACTION_SCROLL:
+				// Process scroll wheel movement:
+				float yDistance = MotionEventCompat.getAxisValue(event, MotionEvent.AXIS_VSCROLL);
+				boolean mouseReport = ((vt320) bridge.buffer).isMouseReportEnabled();
+				if (mouseReport) {
+					int row = (int) Math.floor(event.getY() / bridge.charHeight);
+					int col = (int) Math.floor(event.getX() / bridge.charWidth);
+
+							((vt320) bridge.buffer).mouseWheel(
+											yDistance > 0,
+											col,
+											row,
+											(event.getMetaState() & KeyEvent.META_CTRL_ON) != 0,
+											(event.getMetaState() & KeyEvent.META_SHIFT_ON) != 0,
+											(event.getMetaState() & KeyEvent.META_META_ON) != 0);
+					return true;
+				} else if (yDistance != 0) {
+					int base = bridge.buffer.getWindowBase();
+					bridge.buffer.setWindowBase(base - Math.round(yDistance));
+					return true;
+				}
+			}
+		}
+		return super.onGenericMotionEvent(event);
+	}
+
+	private void copyBufferToText() {
```

Comment:
small nit: The word "copy" threw me off for a minute here because I was looking for text copying. I'd prefer "refreshTextFromBuffer" or something.